### DISCUSSION TITLE
How do I set  compiling options or environment variables to specific dependencies.

### DISCUSSION THREAD

**Discussion by Ericsii** [prefix-dev/pixi#2759] [Q&A] [answered]
- Can I add a CMake project git repo as a dependency?
I read this tutorial page https://pixi.sh/latest/build/cpp_package, and I think this can be done just by adding an item in [dependencies].
- How can I change the compiling options in a CMake project? I didn't see any instructions here https://pixi.sh/latest/reference/pixi_manifest/#the-dependencies-tables. Or can the environment variables be modified while building a dependency?

**Comment by ruben-arts** [comment, marked as answer]
The dependencies repositories do need something that `pixi` can read. So either a `pixi.toml` `pyproject.toml` or a `recipe.yaml` to know how to build the project. For sending arguments to the builds, we're still in the progress of designing that. Currently, that would be easiest through a `recipe.yml` which you can modify on the go.

**Comment by lucascolley** [comment]
https://prefix-dev.github.io/pixi-build-backends/backends/pixi-build-cmake/#extra-args
